src/Contao/View/Contao2BackendView/TreeView.php

```diff
@@ -584,12 +588,24 @@ public function handleAjaxCall()
 
         switch ($input->getValue('action')) {
             case 'DcGeneralLoadSubTree':
+                // TODO load sub as route (entry point).
                 \header('Content-Type: text/html; charset=' . Config::get('characterSet'));
                 echo $this->ajaxTreeView(
                     $input->getValue('id'),
                     $input->getValue('providerName'),
                     $input->getValue('level')
                 );
+
+                // TODO START give the response back or use own entry point for ajax call.
+                // The problem is the session will be update by event kernel.response.
+                $session = System::getContainer()->get('session');
+                $sessionBag = $session->getBag('contao_backend')->all();
+
+                $user = BackendUser::getInstance();
+
+                Database::getInstance()->prepare('UPDATE tl_user SET session=? WHERE id=?')
+                    ->execute(\serialize($sessionBag), $user->id);
+                // TODO END
```

Comment:
Isn't this done in `BackendUser::__destruct()`?

Reply:
The magic method destruct is removed.

Before:
https://github.com/contao/core-bundle/blob/7c452603ae23038b0cd42a54f7b64a494b476a46/src/Resources/contao/classes/BackendUser.php#L112

After:
https://github.com/contao/core-bundle/blob/7b1d8995253edecdd71a11ee4d9eda1f861b009a/src/Resources/contao/classes/BackendUser.php#L112

Reply:
But `storeSession()` is called instead, so we don't have to store the session in this case I suppose (we are using hardcoded `tl_user` here anyway, which is bad).

Reply:
The right solution is to design the conversion in such a way that you can do without the exit. So that the subsequent events of the kernel are executed.

Reply:
So long this is a temporary solution as I described it in the source code.